hermes-console-vue/src/composables/search/useSearch.ts

```diff
@@ -0,0 +1,92 @@
+import { querySubscriptions, queryTopics } from '@/api/hermes-client';
+import { ref } from 'vue';
+import type { Ref } from 'vue';
+import type { Subscription } from '@/api/subscription';
+import type { Topic } from '@/api/topic';
+
+export interface UseSearch {
+  topics: Ref<Topic[] | undefined>;
+  subscriptions: Ref<Subscription[] | undefined>;
+  queryTopicsFn: (filter: SearchFilter, pattern: string) => void;
+  querySubscriptionsFn: (filter: SearchFilter, pattern: string) => void;
+  loading: Ref<boolean>;
+  error: Ref<UseSearchErrors>;
+}
+
+export interface UseSearchErrors {
+  fetchError: Error | null;
+}
+
+export enum SearchFilter {
+  NAME,
+  OWNER,
+  ENDPOINT,
+}
+
+export function useSearch(): UseSearch {
+  const topics = ref<Topic[]>();
+  const subscriptions = ref<Subscription[]>();
+  const error = ref<UseSearchErrors>({
+    fetchError: null,
+  });
+  const loading = ref(false);
+
+  const queryTopicsFn = async (filter: SearchFilter, pattern: string) => {
+    try {
+      loading.value = true;
+      const query = buildQuery(filter, pattern);
+      topics.value = (await queryTopics(query)).data;
+    } catch (e) {
+      console.error(e);
+      error.value.fetchError = e as Error;
+    } finally {
+      loading.value = false;
+    }
+  };
+
+  const querySubscriptionsFn = async (
+    filter: SearchFilter,
+    pattern: string,
+  ) => {
+    try {
+      loading.value = true;
+      const query = buildQuery(filter, pattern);
+      subscriptions.value = (await querySubscriptions(query)).data;
+    } catch (e) {
+      console.error(e);
```

Comment:
Don't use console.error in production code. If we want to console.log something, we should make some implementation of logger, that would check for env and decide if this should be logged or not. For now, we can omit logging

Reply:
fixed

Reply:
removed